conn.go

```diff
@@ -118,12 +118,12 @@ func (c *Conn) Streams() []*Stream {
 // Close closes this connection
 func (c *Conn) Close() error {
 	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
 	if c.closed {
+		c.closeLock.Unlock()
```

Comment:
making them wait is the whole problem i'm trying to avoid, lol. The code in your example is functionally equivalent to the code causing the issue we have now.